### PR TITLE
Remove dependency for node integration tests from e2e fw

### PR DESCRIPTION
**Pull request by gavinfish**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Remove node integration tests' dependency on e2e test framework.

**Which issue(s) this PR fixes**:

Xref #89185

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
none
```

### CODE REVIEW

File: test/integration/node/lifecycle_test.go [hunk 1]
```diff
@@ -297,3 +303,30 @@ func TestTaintBasedEvictions(t *testing.T) {
 		})
 	}
 }
+
+// WaitForPodCondition waits a pods to be matched to the given condition.
+func WaitForPodCondition(c clientset.Interface, ns, podName, desc string, timeout time.Duration, condition podCondition, t *testing.T) error {
```
Comment (tanjunchen) [inline test/integration/node/lifecycle_test.go]: if it is only used here,private is better

File: test/integration/node/lifecycle_test.go [hunk 2]
```diff
@@ -297,3 +303,30 @@ func TestTaintBasedEvictions(t *testing.T) {
...
+			if apierrors.IsNotFound(err) {
+				t.Logf("Pod %q in namespace %q not found. Error: %v", podName, ns, err)
+				return err
+			}
+			t.Logf("Get pod %q in namespace %q failed, ignoring for %v. Error: %v", podName, ns, poll, err)
+			continue
+		}
+		// log now so that current pod info is reported before calling `condition()`
+		t.Logf("Pod %q: Phase=%q, Reason=%q, readiness=%t. Elapsed: %v",
+			podName, pod.Status.Phase, pod.Status.Reason, podutil.IsPodReady(pod), time.Since(start))
```
Comment (tanjunchen) [inline test/integration/node/lifecycle_test.go]: i think we should think about this   `podutil.IsPodReady` , waiting for @oomichi  confirmation. :smile:

File: test/integration/node/lifecycle_test.go (hunk 1 above)
Comment (gavinfish) [reply test/integration/node/lifecycle_test.go]: emm.. it seems my refactor is not be saved.

File: test/integration/node/lifecycle_test.go (hunk 2 above)
Comment (oomichi) [reply test/integration/node/lifecycle_test.go]: I cannot follow @tanjunchen comment here..
Can you clarify your comment?

File: test/integration/node/lifecycle_test.go [hunk 3]
```diff
@@ -297,3 +303,29 @@ func TestTaintBasedEvictions(t *testing.T) {
 		})
 	}
 }
+
+// waitForPodCondition waits a pods to be matched to the given condition.
+func waitForPodCondition(c clientset.Interface, ns, podName, desc string, timeout time.Duration, condition podCondition, t *testing.T) error {
```
Comment (oomichi) [inline test/integration/node/lifecycle_test.go]: nit: This function name is duplicated with test.waitForPodCondition which is passed to the above `desc string`.
That is a little less code readability, but that is not matter actually.
Maybe it is better to rename test.waitForPodCondition to the other one with different PR.

**Review by oomichi — COMMENTED** [review]
Thanks for doing this.
I'd like to see @tanjunchen comment clearly.

File: test/integration/node/lifecycle_test.go (hunk 3 above)
Comment (gavinfish) [reply test/integration/node/lifecycle_test.go]: Sure, I will follow up.

File: test/integration/node/lifecycle_test.go (hunk 2 above)
Comment (tanjunchen) [reply test/integration/node/lifecycle_test.go]: sorry,i misunderstand. `podutil.IsPodReady` can be used here (test/integration)

**Pull request merged**